cEP-0001.md

```diff
@@ -4,38 +4,101 @@ Decision Making in coala
+and how they can vote on different levels of decisions.
 
-This cEP describes a voting based decision making level with several levels
-of decisions. Higher levels need more people to agree with the decision.
 
 Roles
 -----
 
-The following roles may be involved in decisions:
+The following roles can be held by members of the coala community
+
+### Newcomers
+
+A newcomer is someone that was just introduced to the coala project and
+community and begins to contribute to it. This rank mainly represents the
+learning phase of the contribution process.
+
+Every community member that started the [newcomer guide](coala.io/newcomer) is
+declared a newcomer by invitation into the GitHub organisation. Newcomers can
+work on issues with an assignment.
+
+### Developers
+
+A developer is someone that showed capable of contributing to coala. They are
+involved on a normal level and help with contributions and reviews on a
+semi-regular basis.
+
+When a newcomer finished the newcomer guide they can ask a maintainer to be
+promoted developer. A developer has all the rights of a newcomer but can be an
+official coala team member and can use cobot to label issues and pull requests.
+Developers also have voting rights on A level decisions.
+
+If the contributions of a developer show the lack of fundamental understanding
+of the coala workflow they can be demoted to the newcomer rank with a B level
+decision. The promotion to developer must be granted after a renewed completion
+of the newcomer guide.
+A developer can choose to be demoted newcomer at any time.
+
+### Maintainers
 
-### Members
+A Maintainer is someone that showed persistence in contributions and or an
+increased quality compared to a developer. They offer experience and passion for
+the project and are granted full voting rights due to their achivements.
+
+A developer can be promoted maintainer with a B level decision. This decision
+does not have to be initiated by a maintainer so it is valid for a developer to
+propose himself for a promotion.
+A maintainer has all the rights a developer has. Additionally maintainers have
+write access to coala repositories, can promote newcomers and developers and
+have voting rights on B and C level decisions.
+
+The two important aspects of the maintainer rank are the voting rights for B and
+C level decisions and the write access. While the write access mostly is a
+convenience feature that is granted to long time contributers as a quality of
+life improvement and sign of trust, the voting rights also show the valuation
+of ones knowledge and optinions.
+It is due to these considerations, that it is valid to promote contributers to
+maintainers that did not finish the newcomer guide. This enables non coding
+contributors to add their input to important decisions.
+
+The demotion from maintainer to developer represents a C level decision. It
+should only be done if a maintainer has shown inactivity for prolonged periods
+of time, half a year and upwards.
```

Comment:
uhm if we need a c level decision for demoting members that's not gonna help our "we have inactive members" problem